server/src/main/java/org/opensearch/cluster/decommission/DecommissionStatus.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.decommission;
+
+/**
+ * An enumeration of the states during decommissioning and recommissioning.
+ */
+public enum DecommissionStatus {
+    /**
+     * Decommission process is initiated
+     */
+    DECOMMISSION_INIT("decommission_init"),
+    /**
+     * Decommission process has started, decommissioned nodes should be weighed away
+     */
+    DECOMMISSION_IN_PROGRESS("decommission_in_progress"),
+    /**
+     * Decommissioning awareness attribute completed
+     */
+    DECOMMISSION_SUCCESSFUL("decommission_successful"),
+    /**
+     * Decommission request failed
+     */
+    DECOMMISSION_FAILED("decommission_failed"),
+    /**
+     * Recommission request received, recommissioning process has started
+     */
+    RECOMMISSION_IN_PROGRESS("recommission_in_progress"),
+    /**
+     * Recommission request failed. No nodes should fail to join the cluster with decommission exception
+     */
+    RECOMMISSION_FAILED("recommission_failed");
```

Comment:
lets remove this. Just the absence of decommission metadata is `recommission`. We will never now when recommission is complete or not.

Reply:
Makes sense. For the initial phase, we don't need it. Will remove